openmc/material.py

```diff
@@ -282,15 +275,49 @@ def fissionable_mass(self) -> float:
 
     @property
     def decay_photon_energy(self) -> Optional[Univariate]:
-        atoms = self.get_nuclide_atoms()
+        warnings.warn(
+            "The 'decay_photon_energy' property has been replaced by the "
+            "get_decay_photon_energy() method and will be removed in a future "
+            "version.", FutureWarning)
+        return self.get_decay_photon_energy(0.0)
+
+    def get_decay_photon_energy(self, clip_tolerance: float = 1e-6) -> Optional[Univariate]:
+        """Return energy distribution of decay photons from unstable nuclides.
+
+        Parameters
+        ----------
+        clip_tolerance : float
+            Maximum fraction of integral of the product of `x` and `p` for
+            discrete distributions that will be discarded.
+
+        Returns
+        -------
+        Decay photon energy distribution. The integral of this distribution is
+        the total intensity of the photon source in [Bq].
+
+        """
         dists = []
         probs = []
-        for nuc, num_atoms in atoms.items():
+        for nuc, num_atoms in self.get_nuclide_atoms().items():
             source_per_atom = openmc.data.decay_photon_energy(nuc)
             if source_per_atom is not None:
                 dists.append(source_per_atom)
                 probs.append(num_atoms)
-        return openmc.data.combine_distributions(dists, probs) if dists else None
+
+        # If no photon sources, exit early
+        if not dists:
+            return None
+
+        # Clip low-intensity values in discrete spectra
+        combined = openmc.data.combine_distributions(dists, probs)
+        if isinstance(combined, Discrete):
+            combined.clip(clip_tolerance, inplace=True)
+        elif isinstance(combined, Mixture):
```

Comment:
Is it worth adding this logic here as a `clip` method on the `Mixture` class? and just returning `combined.clip(clip_tolerance, inplace=True)`

Reply:
I thought about that too actually. I can add it to `Mixture` but to have a single `combined.clip(...)` call we would need to add it to `Tabular` too where it doesn't make sense, which is why I decided to just put the logic for the going through the mixture distributions here. Let me know if you think it's still worth it to add that to `Mixture` (would save 2 lines of code here).

Reply:
ok no big deal either way, but if it was on `Mixture` could we just do 
```python
if hasattr(combined, clip):
    return combined.clip(...)
return combined
```
I think it might still be worth having on `Mixture` distributions themselves just for the functionality. Regardless of the small line count differences what do you think?